spec/models/address_spec.rb

```diff
@@ -0,0 +1,38 @@
+require 'rails_helper'
+
+RSpec.describe Address, type: :model do
+  subject { build(:address) }
+
+  it 'is valid with all valid attributes' do
+    expect(subject).to be_valid
+  end
+
+  it 'is not valid without a building name' do
+    subject.address_line_one = ''
+    expect(subject).to_not be_valid
+    expect(subject.errors[:address_line_one]).to include("can't be blank")
```

Comment:
What are you testing here? That the rails method `validates` works?

Reply:
If you don't have any tests in place, you don't really know if or which validations are being performed. Where as it they can be done some other way, that's another question, but I believe these are still valid tests.

Reply:
My main point, is that if you start adding specs that just test rails functionality, and not the app specific logic, you end up with a lot of unnecessary specs and they just act as noise to make it more difficult to identify the important specs.

If you want to know which validations are applied to a model - go and look in the model code.

In general - if a validation is standard out-of-the-box rails, testing it is testing rails and not your code. If you create a custom validator (for example with `validate :my_custom_validator`) that needs testing.

The same with joins. `has_many :foos` doesn't need testing. Testing it just tests how `has_many` works and is therefore unnecessary. However, `has_many :bars, -> { where('bars.names LIKE "%thimble"') }` that's got app specific logic and needs testing.